code/game/objects/structures/prop.dm

```diff
@@ -210,10 +210,109 @@
 	mouse_opacity = 0
 	layer = LADDER_LAYER
 
+/obj/structure/prop/mainship/supermatter //functionally inert, but will consume mobs and objects
+	name = "supermatter crystal"
+	desc = "A strangely translucent and iridescent crystal."
+	icon = 'icons/Marine/mainship_props64.dmi'
+	icon_state = "darkmatter"
+	layer = LADDER_LAYER
+	light_range = 4
+	resistance_flags = RESIST_ALL //no delaminations here
+
+
+/obj/structure/prop/mainship/supermatter/proc/Consume(atom/movable/consumed_object)
+	if(isliving(consumed_object))
+		var/mob/living/consumed_mob = consumed_object
+		consumed_mob.dust()
+	else if(isobj(consumed_object))
+		qdel(consumed_object)
```

Comment:
use early return please

Reply:
I don't understand this one. If I did "if(!isliving(consumed_object) return" then it would skip the check/qdel of thrown inanimate objects, which is intended behavior.